source/common/router/config_impl.h

```diff
@@ -367,8 +367,12 @@ class RouteMatcher {
 
 private:
   const VirtualHostImpl* findVirtualHost(const Http::HeaderMap& headers) const;
+  const VirtualHostImpl* findWildcardVirtualHost(const std::string& host) const;
 
   std::unordered_map<std::string, VirtualHostSharedPtr> virtual_hosts_;
+  // std::greater as a minor optimization to iterate from more to less specific
+  std::map<int64_t, std::unordered_map<std::string, VirtualHostSharedPtr>, std::greater<int64_t>>
```

Comment:
instead of using an unordered_map here, would it be better to use a vector of pairs? I kind of suspect that the map will almost always contain a single entry, and the iteration is probably going to be faster? Not sure, but thought I would throw it out there.